io-functions-pay-portal/GetPaymentInfo/handler.ts

```diff
@@ -8,33 +8,64 @@ import {
   wrapRequestHandler
 } from "io-functions-commons/dist/src/utils/request_middleware";
 import {
-  IResponseErrorNotFound,
   IResponseSuccessJson,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
 
 import { PaymentRequestsGetResponse } from "../generated/definitions/PaymentRequestsGetResponse";
+import { IApiClient} from "../clients/pagopa";
+import { withApiRequestWrapper } from "../utils/api";
+import { getLogger, ILogger } from "../utils/logging";
+import { ErrorResponses } from "../utils/responses";
+import { identity } from "fp-ts/lib/function";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { Context } from "@azure/functions";
 
-type IHttpHandler = () => Promise<
-  IResponseSuccessJson<PaymentRequestsGetResponse> | IResponseErrorNotFound
+type IGetPaymentInfoHandler = (
+    context: Context,
+    rptId : string
+) => Promise<
+  IResponseSuccessJson<PaymentRequestsGetResponse> | ErrorResponses
 >;
 
-export function HttpHandler(): IHttpHandler {
-  return async () => {
-    return ResponseSuccessJson({
-      codiceContestoPagamento: "testcode",
-      importoSingoloVersamento: 101
-    } as PaymentRequestsGetResponse);
-  };
-}
+const logPrefix = "GetPaymentInfoHandler";
 
-export function HttpCtrl(): express.RequestHandler {
-  const handler = HttpHandler();
+const getPaymentInfoTask = (
+  logger: ILogger,
+  apiClient: IApiClient,
+  rptId : string
+): TaskEither<ErrorResponses, PaymentRequestsGetResponse> =>
+  withApiRequestWrapper(
+    logger,
+    (): any =>
+      apiClient.getPaymentInfo({ rpt_id_from_string : rptId }) 
+      ,
+    200
+  );
 
+export function GetPaymentInfoHandler(pagoPaClient : IApiClient): IGetPaymentInfoHandler {
+  return (context, rptId) => 
+    getPaymentInfoTask(
+      getLogger(context, logPrefix, "GetPaymentInfo"),
+      pagoPaClient, 
+      rptId)
+      .map(myPayment =>
+        ResponseSuccessJson(myPayment)
+      )
+      .fold<IResponseSuccessJson<PaymentRequestsGetResponse> | ErrorResponses>(
+      identity,
+      identity
+      )
+      .run();
+}
+
+export function GetPaymentInfoCtrl(pagoPaClient : IApiClient): express.RequestHandler {
+  const handler = GetPaymentInfoHandler(pagoPaClient);
   const middlewaresWrap = withRequestMiddlewares(
     ContextMiddleware(),
     RequiredParamMiddleware("rptId", t.string)
```

Comment:
There is an io-ts type available in io-pagopa-commons that will parse an RptId from a string

Reply:
Tnx, i'll use this one `export declare const RptIdFromString` which I believe you are referencing

Reply:
Yes, that's the one